src/GeoInterfaceRFC.jl

```diff
@@ -5,47 +5,84 @@ struct Geometry <: AbstractGeometry end
 
 abstract type AbstractPoint <: AbstractGeometry end
 struct Point <: AbstractPoint end
-abstract type AbstractLineString <: AbstractGeometry end
+
+abstract type AbstractCurve <: AbstractGeometry end
+abstract type AbstractLineString <: AbstractCurve end
 struct LineString <: AbstractLineString end
-abstract type AbstractPolygon <: AbstractGeometry end
+struct Line <: AbstractLineString end  # LineString with just two points
+struct LinearRing <: AbstractLineString end  # Closed LineString
+
+abstract type AbstractSurface <: AbstractGeometry end
+abstract type AbstractPolygon <: AbstractSurface end
+abstract type AbstractPolyHedralSurface <: AbstractSurface end
 struct Polygon <: AbstractPolygon end
+struct Triangle <: AbstractPolygon end  # Polygon with just three points
+struct Box <: AbstractPolygon end  # Polygon with just 4 points
+struct PolyHedralSurface <: AbstractPolyHedralSurface end
+struct TIN <: AbstractPolyHedralSurface end
 
 abstract type AbstractMultiPoint <: AbstractGeometry end
 struct MultiPoint <: AbstractMultiPoint end
-abstract type AbstractMultiLineString <: AbstractGeometry end
+
+abstract type AbstractMultiCurve <: AbstractGeometry end
+abstract type AbstractMultiLineString <: AbstractMultiCurve end
 struct MultiLineString <: AbstractMultiLineString end
-abstract type AbstractMultiPolygon <: AbstractGeometry end
+
+abstract type AbstractMultiSurface <: AbstractGeometry end
+abstract type AbstractMultiPolygon <: AbstractMultiSurface end
 struct MultiPolygon <: AbstractMultiPolygon end
 
 abstract type AbstractGeometryCollection <: AbstractGeometry end
 struct GeometryCollection <: AbstractGeometryCollection end
 
 # All Geometries
-function geomtype(x::Any)
+function geomtype(x::T) where T
```

Comment:
Is there any reason you prefer `(x::T) where T` over `(x::Any)`? I suspect the `::Any` was there to make it extra explicit that it is a fallback, but we can also just leave it out `(x)`.